### PR TITLE
Disable editor preview

### LINKED ISSUE #113: Add possibility to disable Preview panel
Sometimes, The preview panel do not have a real interest because we do not have markdown field. So I could be great to disable it in config file for a collection

### PR DESCRIPTION
**Pull request by josephearl**
**- Summary**

Fixes #113. Adds ability to disable the editor preview for a collection.

**- Test plan**

Added setting to example `config.yml` for settings collection. Tested with `npm start`.

**- Description for the changelog**

Adds `editor.disable_preview` option to collection. If set to `true`, the editor preview will not be displayed and the button to show the preview will not be displayed.

Not sure it was the best option name so let me know if there's a better one.

### CODE REVIEW

File: src/components/EntryEditor/EntryEditor.js [hunk 1]
```diff
@@ -53,9 +53,15 @@ class EntryEditor extends Component {
...
+    const disablePreview = collection.getIn(['editor', 'disable_preview'], false);
+
+    const togglePreviewButton = (
+      <Button className={styles.previewToggle} onClick={this.handleTogglePreview}>Toggle Preview</Button>
+    );
+
     const editor = (
       <div className={controlClassName}>
-        <Button className={styles.previewToggle} onClick={this.handleTogglePreview}>Toggle Preview</Button>
+        { !disablePreview ? togglePreviewButton : null }
```
Comment (Benaiah) [inline src/components/EntryEditor/EntryEditor.js]: This involves too many repeated negations for my taste - "not disable preview" is doubly negated and unnecessarily hard to follow.

File: src/components/EntryEditor/EntryEditor.js [hunk 2]
```diff
@@ -97,7 +103,7 @@ class EntryEditor extends Component {
 
     return (
       <div className={styles.root}>
-        { this.state.previewVisible ? editorWithPreview : editor }
+        { !disablePreview && this.state.previewVisible ? editorWithPreview : editor }
```
Comment (Benaiah) [inline src/components/EntryEditor/EntryEditor.js]: Likewise.

File: src/components/EntryEditor/EntryEditor.js [hunk 3]
```diff
@@ -53,9 +53,15 @@ class EntryEditor extends Component {
     const controlClassName = `${ styles.controlPane } ${ this.state.showEventBlocker && styles.blocker }`;
     const previewClassName = `${ styles.previewPane } ${ this.state.showEventBlocker && styles.blocker }`;
 
+    const disablePreview = collection.getIn(['editor', 'disable_preview'], false);
```
Comment (Benaiah) [inline src/components/EntryEditor/EntryEditor.js]: Tying into my comments about the change of the setting name, I would change this to something like `collectionHasPreview` to make it more clear where the source of this value is, particularly since there's already `this.state.previewVisible`.

Additionally, if the setting name changes to `editor.preview`, this should default to `true`.

File: example/config.yml [hunk 4]
```diff
@@ -30,6 +30,8 @@ collections: # A list of collections the CMS should be able to edit
 
   - name: "settings"
     label: "Settings"
+    editor:
+      disable_preview: true
```
Comment (Benaiah) [inline example/config.yml]: I suggest changing the setting to "editor.preview" and making it default to "true". Setting a property with "disable" in the name to "true" is an unnecessary negation of the actual value we want to know: *does this collection use previews?*

Comment (josephearl) [inline example/config.yml]: :+1:

**Comment by josephearl** [comment]
@Benaiah updated -- I used `editor.preview` in the config with a default of `true` as suggested. In the component I went with `collectionPreviewEnabled` as the variable name.

**Comment by josephearl** [comment]
@erquhart @Benaiah how do you prefer to merge PRs? 'Create a merge commit' or 'Rebase and merge'?

**Comment by erquhart** [comment]
Create a merge commit.

**Comment by Benaiah** [comment]
@josephearl LGTM, thanks for getting that fixed.

**Comment by josephearl** [comment]
Great, thanks guys 👍

**Comment by attharmirza** [comment]
@josephearl thank you so much! I really needed this.
